### DISCUSSION TITLE
Domotiser entièrement un radiateur éléctrique dans Home assistant - Haade.fr

### DISCUSSION THREAD

**Discussion by giscus** [haade-administrator/haade.fr#19] [Comments]
# Domotiser entièrement un radiateur éléctrique dans Home assistant - Haade.fr

Article complet pour domotiser un radiateur électrique sur fil pilote et contrôler l’ordre confort/hors gel et arrêt avec inversion de commande on/off et intégration dans lovelace d’un thermostat générique, l’ensemble te permettra de faire de réels économies d’énergies

https://haade.fr/fr/blog/domotiser-entierement-un-radiateur-electrique-dans-home-assistant

**Comment by legrand76** [comment]
Bonjour,

Savez vous si depuis une mise a jour de Home Assistant cela fonctionne toujours ?
Car chez moi ça ne fonctionne pas.
J'avais réaliser le montage en Mai et ça focntionnait

**Reply by haade-administrator** [reply]
au dernières nouvelles ça fonctionne je vais faire un tour sur une install et te tiens au jus

**Reply by legrand76** [reply]
Merci d'avance 
Lorsque j'active la chauffe cela me marque inactif et aucun changement sur le switch

**Reply by legrand76** [reply]
Salut,
J'ai trouvé le défaut 
Il s'agit de la ligne   cold_tolerance: 10
Une fois supprimé ça fonctionne

Merci pour ce tuto

**Reply by haade-administrator** [reply]
ah oui c'est vrai je dois la supprimer ça t'indique une différence de 10 pour s'enclancher

**Reply by haade-administrator** [reply]
je le fait de suite

**Comment by HA-FAMILLEPHC** [comment]
Bonjour je débute sur la gestion du chauffage avec homeassistant. 
Lorsque que l'on a plusieurs  radiateurs   est ce qu'il faut créer autant de fichiers pour inverser les commandes du switch, qu'il y a de radiateur, ou est-ce que l'on peut faire un seul fichier regroupant tout ?
Même chose pour le fichier climat.
Merci pour votre aide

**Reply by legrand76** [reply]
Bonjour,
Oui moi j'ai fais un fichier switch.yaml + 1 climate.yaml pour 4 radiateurs
Et j'ai inclus ces deux lignes dans le fichier config.yaml

#Chauffage
switch: !include switch.yaml
climate: !include_dir_merge_list climate/

**Comment by Homeassistant17** [comment]
Bonjour.

Il me semble que les sens des diodes sont inversés sur les schémas.

**Comment by mathieucarbou** [comment]
> # maintient en chauffe au moins 3 minutes si le thermostat ne reçoit pas de commande

Attention, ce n'est pas vraiment vrai.

Le `keep_alive` est juste un paramètre qui dit à HA d'envoyer la commande "allumer prise" toutes les 3 minutes dans ta configuration. 

Le `keep_alive` sert uniquement dans le cas ou l'on met en place une sécurité par timer, par exemple dès que la prise du chauffage est allumée, un timer dans la prise se déclenche pour l'éteindre au bout de 30min par sécurité. Et le `keep_alive` va s'assurer que la prise ne s'éteigne pas justement. Si le thermostat ne reçoit plus de commande (HA dead par exemple), le chauffage ne s'éteindra pas, SAUF si un timer a été configuré dans sa prise.

**Comment by Surfnuma05** [comment]
Bonjour,
Cela fait maintenant plusieurs jours que je me bats avec les différentes lignes de code et je n'arrive pas à comprendre pourquoi je n'arrive pas à piloter mon radiateur (switch manuellement tout d'abord et programmation ensuite). 

Bien que j'ai suivi les différentes étapes énoncées dans l'article, rien n'y fait. 
ZHA : 2023.11.3
Nom de mon thermostat : TH-Bureau
Nom de module zigbee de pilotage : Pilotage Radiateur Bureau

Voilà ce que j'ai saisi dans les différents fichiers indiqués
switch: !include switch.yaml  dans le fichier config
Puis dans un fichier switch.yaml à la racine du dossier j'ai ajouté le code suivant en respectant la case et les espaces: 
- platform: template
  switches:
    bureau_inverted:
      friendly_name: Radiateur bureau inversé
      value_template: "{{ is_state('switch.Pilotage_Radiateur_Bureau', 'off') }}"
      turn_on:
        service: switch.turn_off
        data:
          entity_id: switch.Pilotage_Radiateur_Bureau
      turn_off:
        service: switch.turn_on
        data:
          entity_id: switch.Pilotage_Radiateur_Bureau

En suite j'ai ajouté la ligne climate au fichier de configuration.yaml
climate: !include_dir_merge_list climate/

puis j'ai créé un fichier generic_thermostat.yaml et collé le code ci-dessous : 
- platform: generic_thermostat
  name: Température Bureau
  heater: switch.bureau_inverted
  target_sensor: sensor.th_bureau_temperature
  min_temp: 16
  max_temp: 18
  ac_mode: false
  target_temp: 17
  cold_tolerance: 0
  hot_tolerance: 0.3
  initial_hvac_mode: "off"
  away_temp: 12
  precision: 0.5
  min_cycle_duration:
    seconds: 5
  keep_alive:
    minutes: 3


Je sauvegarde à chaque fois, je vérifie et je redémarra puis une fois redémarré j'ajoute la carte du thermostat généric. et c'est la que rien ne marche. 

Je n'arrive pas à comprendre pourquoi je n'y arrive pas. 

Auriez vous quelques instant pour m'expliquer au novice que je suis 

Merci

**Comment by fossetg** [comment]
Bonsoir,
Merci pour ce tuto éclairant.
Je l'ai mis en oeuvre avec zbmini sous HA.
J'ai toutefois voulu garder un accès manuel au fonctionnement du radiateur en ajoutant un interrupteur bistable entre S1 et S2.
Le problème : Si je manipule le module via HA, le radiateur s'allume ou s'éteint. Si j'en fait autant avec l'interrupteur, même résultat mais l'état de module ne se synchronise pas dans HA.
Une idée ? Un problème de puissance de signal ? Mon émetteur Zigbee est au rez de chaussée.
Pourtant, les détecteurs d'ouverture/vibration fonctionnent parfaitement.
Merci d'avance
Bonne soirée

**Reply by haade-administrator** [reply]
Salut le retour d'état devrait remonter aurais-tu un scénario qui empêche celà ?

**Reply by fossetg** [reply]
Bonsoir, merci pour ton retour rapide.
Pas de scénario particulier. Je suis sur une installation "vierge" de HA avec zigbee2mqtt sur RPI 5. Je teste divers dispositifs (ouverture/vibration, lumière, volet, présence, VMC... le tout principalement à partir de modulés sonoff à part pour le détecteur ouverture/vibration lexman) avant de monter l'ensemble de l'installation. Est ce que le signal pourrait être suffisant pour que je puisse actionner le module à partir HA mais que le retour d'état du module ne remonte pas si je manipule l'interrupteur ? Bonne soirée

**Reply by fossetg** [reply]
Pour infos, cela ne marche pas avec ZBmini Extrème sans neutre. Je suppose que c'est normal.
